src/app/components/package-list/package-list.component.ts

```diff
@@ -0,0 +1,90 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, forwardRef, Input, OnInit, Output } from '@angular/core';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { takeUntilDestroy } from '../../operators';
+import { RegistryData } from '../../services/npm-registry/npm-registry.model';
+
+enum SelectKeydown {
+	SPACE = 'Space',
+	SPACE_2 = ' ',
+	ENTER = 'Enter',
+}
+
+const POSSIBLE_KEYDOWNS = [SelectKeydown.ENTER, SelectKeydown.SPACE, SelectKeydown.SPACE_2];
+
+@Component({
+	selector: 'app-package-list',
+	templateUrl: './package-list.component.html',
+	styleUrls: ['./package-list.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => PackageListComponent),
+			multi: true,
+		},
+	],
+})
+export class PackageListComponent implements OnInit, ControlValueAccessor {
+	@Output() removePackageName: EventEmitter<string> = new EventEmitter<string>();
+	@Input() registryData: RegistryData[] | null = [];
+
+	readonly selectedPackageNames = new FormControl<string[]>([], {
+		nonNullable: true,
+	});
+
+	private selectedPackageNames$ = this.selectedPackageNames.valueChanges.pipe(takeUntilDestroy());
```

Comment:
takeUntilDestroy$() only when subscribing:

obs$.pipe(
  takeUntilDestroy$()
).subscribe();

Reply:
Sadly this operator has to be used before onInit() hook otherwise you get an error
`ERROR Error: NG0203: inject() must be called from an injection context such as a constructor, a factory function, a field initializer, or a function used with `EnvironmentInjector#runInConte` 
Reference: https://angular.io/errors/NG0203